Restore .NET Framework config docs + 1

guardrex opened the pull request:
Fixes #1981 

* Restores basic config docs
* Restores [.NET Framework 4 Migration Issues](https://msdn.microsoft.com/en-us/library/ee941656(v=vs.100).aspx)

Commit:
Restore docs

Updates

GuardRex commented:
https://review.docs.microsoft.com/en-us/dotnet-new/articles/framework/migration-guide/net-framework-4-migration-issues?branch=pr-en-us-2081

Commit:
Updates

guardrex committed:
Merge branch 'master' into guardrex/restore-docs

guardrex committed:
Merge branch 'master' into guardrex/restore-docs

Commit:
Updates

guardrex committed:
Merge branch 'master' into guardrex/restore-docs

Commit:
MSDN link conversion

GuardRex commented:
@mairaw The script was only ~80% correct in link swaps. Do you prefer to go with the MSDN version (~386 MSDN links in it) or go with the script-corrected version and have me manually fix these ~60 👇 bad xref links?

> Invalid cross reference "&lt;xref:PagesSection.ControlRenderingCompatibilityVersion&gt;", "&lt;xref:MenuItem.PopOutImageUrl&gt;", "&lt;xref:Menu.StaticPopOutImageUrl&gt;", "&lt;xref:Menu.DynamicPopOutImageUrl&gt;", "&lt;xref:HttpCachePolicy.SetOmitVaryStar&gt;", "&lt;xref:HttpResponse.AppendHeader&gt;", "&lt;xref:HttpRequest.FilePath&gt;", "&lt;xref:HttpRequest.AppRelativeCurrentExecutionFilePath&gt;", "&lt;xref:HttpRequest.CurrentExecutionFilePath&gt;", "&lt;xref:HttpRequest.PathInfo&gt;", "&lt;xref:TimeZoneInfo.Local&gt;", "&lt;xref:DateTime.Now&gt;", "&lt;xref:CultureAndRegionInfoBuilder.CurrencyEnglishName&gt;", "&lt;xref:DateTimeFormatInfo.LongDatePattern&gt;", "&lt;xref:DateTimeFormatInfo.PMDesignator&gt;", "&lt;xref:AssemblyName.HashAlgorithm&gt;", "&lt;xref:Assembly.GetReferencedAssemblies&gt;", "&lt;xref:Type.DeclaringType&gt;", "&lt;xref:IList.IndexOf&gt;", "&lt;xref:DataRowView.PropertyChanged&gt;", "&lt;xref:DataRow.RejectChanges&gt;", "&lt;xref:SqlCommand.Prepare&gt;", "&lt;xref:ObjectContext.Detach&gt;", "&lt;xref:ObjectContext.SaveChanges&gt;", "&lt;xref:Exception.InnerException&gt;", "&lt;xref:Window.WindowStyle&gt;", "&lt;xref:Window.AllowsTransparency&gt;", "&lt;xref:CommonDialog.ShowDialog&gt;", "&lt;xref:ItemContainerGenerator.ContainerFromItem&gt;", "&lt;xref:CollectionView.IndexOf&gt;", "&lt;xref:Page.ShowsNavigationUI&gt;", "&lt;xref:UIElement.InvalidateVisual&gt;", "&lt;xref:NavigatingCancelEventArgs.IsNavigationInitiator&gt;", "&lt;xref:DependencyObject.SetCurrentValue&gt;", "&lt;xref:TextBoxBase.Copy&gt;", "&lt;xref:TextBoxBase.Cut&gt;", "&lt;xref:ApplicationCommands.Copy&gt;", "&lt;xref:ApplicationCommands.Cut&gt;", "&lt;xref:ButtonBase.Command&gt;", "&lt;xref:DrawingContext.PushEffect&gt;", "&lt;xref:BitmapSource.DownloadProgress&gt;", "&lt;xref:BitmapSource.DownloadCompleted&gt;", "&lt;xref:BitmapSource.DownloadFailed&gt;", "&lt;xref:BitmapSource.DecodeFailed&gt;", "&lt;xref:InputBinding.Command&gt;", "&lt;xref:InputBinding.CommandParameter&gt;", "&lt;xref:InputBinding.CommandTarget&gt;", "&lt;xref:KeyEventArgs.Key&gt;", "&lt;xref:KeyEventArgs.DeadCharProcessedKey&gt;", "&lt;xref:TreeViewItemAutomationPeer.GetChildrenCore&gt;", "&lt;xref:UIElementAutomationPeer.IsOffscreenCore&gt;", "&lt;xref:MenuItemAutomationPeer.GetChildrenCore&gt;", "&lt;xref:ThumbAutomationPeer.GetClassNameCore&gt;", "&lt;xref:UIElementAutomationPeer.GetChildrenCore&gt;", "&lt;xref:MarkupExtension.ProvideValue&gt;", "&lt;xref:XNode.CreateReader&gt;", "&lt;xref:XElement.Load&gt;", "&lt;xref:XmlReader.ReadContentAs&gt;", "&lt;xref:XmlReader.Create&gt;", "&lt;xref:XmlWriter.WriteEntityRef&gt;", "[Version Compatibility in the .NET Framework](xref:~/docs/framework/migration-guide/version-compatibility.md)".

GuardRex commented:
:point_up: I think that if I just re-run the script adding `%2A` to those that it will knock a bunch of them out, too.

Commit:
Link repairs

Commit:
Repair links

Commit:
Minor updates

GuardRex commented:
@mairaw @rpetrusha Finally! 😅 I think it's ready for :eyes: when the build is finished.

rpetrusha commented on docs/framework/configure-apps/file-schema/assemblybinding-element-for-configuration.md:
```diff
@@ -25,57 +25,59 @@ author: "mcleblanc"
+
+## Example
+
+The following code example shows how to include a configuration file on the local hard disk:
```
We should replace occurrences of the phrase "code example" with "example"

rpetrusha commented on docs/framework/configure-apps/file-schema/clear-element-for-configsections.md:
````diff
@@ -0,0 +1,82 @@
+```xml
+<clear/>
+```
+
````
We should have a Parent element, Child element, and Attributes section

rpetrusha commented on docs/framework/configure-apps/file-schema/clear-element-for-configsections.md:
```diff
@@ -0,0 +1,82 @@
+
+## Example
+
+The following example shows how to use the **\<clear>** element in an application configuration file to clear sections previously defined in the machine configuration file.
```
Maybe make the forward reference here a little bit clearer: This example defines a machine configuration file and an application configuration file and shows...

rpetrusha commented on docs/framework/configure-apps/file-schema/clear-element-for-configsections.md:
````diff
@@ -0,0 +1,82 @@
+             type="System.Configuration.NameValueSectionHandler" />
+  </configSections>
+</configuration>
+```
````
There really should be a `<anotherSection>` section here that has one or two elements defined in it.

rpetrusha commented on docs/framework/configure-apps/file-schema/clear-element-for-custom-2.md:
```diff
@@ -0,0 +1,78 @@
+
+## Example
+
+The following example shows how to use the **\<clear>** element in an application configuration file to clear sections previously defined in the machine configuration file.
```
The same change as previously: The following example defines...

rpetrusha commented on docs/framework/configure-apps/file-schema/configsections-element-for-configuration.md:
```diff
@@ -0,0 +1,66 @@
+[**\<configuration>**](~/docs/framework/configure-apps/file-schema/configuration-element.md)   
+&nbsp;&nbsp;**\<configSections>**
+
+## Child elements
```
Add Parent element and Attributes sections

rpetrusha commented on docs/framework/configure-apps/file-schema/configuration-element.md:
```diff
@@ -26,37 +26,41 @@ author: "mcleblanc"
+
+| Element | Description |
+| ------- | ----------- |
+| [**\<assemblyBinding>**](~/docs/framework/configure-apps/file-schema/assemblybinding-element-for-configuration.md) | Specifies assembly binding policy at the configuration level.|
```
I think it might be useful here in each case to specify the name of the element -- i.e., "| <runtime> settings schema| All elements in the runtime settings schema|"

rpetrusha commented on docs/framework/configure-apps/file-schema/configuration-element.md:
````diff
@@ -26,37 +26,41 @@ author: "mcleblanc"
+</configuration>
+```
+
+## Child elements
````
There should still be Parent and Attribute sections

rpetrusha commented on docs/framework/configure-apps/file-schema/configuration-sections-schema.md:
```diff
@@ -22,33 +19,43 @@ helpviewer_keywords:
+
+# Configuration sections schema
+
+The configuration sections schema contains elements that define custom settings in configuration files.
```
This is a longer list than the original document; none of the additions belong here. There also should be a TOC here that has Configuration Sections Schema with configSections as its child, etc.
It would also be good to figure out what headers to add here, other than See also -- which is the only thing that appears in the mini-TOC (not good).

GuardRex replied:
@rpetrusha I reverted that new content here, but I'm questioning the value of this topic ...

> none of the additions belong here

It was my attempt to make the doc more relevant by showing the entire schema, which doesn't appear anywhere here. I tried to make good on the premise (received by me anyway) that this topic will show all of "elements that define custom settings in configuration files."

Idk what headings to use. If the mini-TOC that results in only one section showing up from *See also* is the main problem, I could roll the single link from *See also* into a sentence on Line 29 and eliminate the *See also* section.

rpetrusha reviewed (changes requested):
@GuardRex, thanks for doing this. I got through configuration-sections-schema.md. I'll try to review the remainder tomorrow.

guardrex committed:
Merge branch 'master' into guardrex/restore-docs

guardrex committed:
Merge branch 'master' into guardrex/restore-docs

GuardRex commented:
@rpetrusha Checking-in ... I think you were called away on high priority work. I was going to wait until the whole PR is reviewed before getting into it; but if you think that will take a while, I can jump in and address what you've identified thus far.

rpetrusha commented on docs/framework/configure-apps/file-schema/section-element.md:
```diff
@@ -0,0 +1,92 @@
+| Attribute | Description |
+| --------- | ----------- |
+| **name**  | Specifies the name of the configuration section. |
+| **type**  | Specifies the name of the configuration section handler class that reads the section from the configuration file. |
```
This needs to have a bit more detail. The **type** value has the syntax "fully-qualified-section-handler-class-name, simple-assembly-name". The simple assembly name is the root filename without the .dll file extension.

rpetrusha commented on docs/framework/configure-apps/file-schema/section-element.md:
```diff
@@ -0,0 +1,92 @@
+
+## Optional attributes
+
+The following attributes are applicable only for ASP.NET applications. The configuration system ignores these attributes when other types of applications are running.
```
when other types of applications are running --> for other application types.

rpetrusha reviewed (commented):
Sorry, @GuardRex, I completely lost track of this one. Here are the remaining comments.

GuardRex commented:
@rpetrusha Thanks much. I'll have the 1st draft on *Create custom templates for dotnet new* done this evening and jump on this feedback next (first thing in the morning).

Commit:
React to feedback

Idk what to do about the headings for
*docs/framework/configure-apps/file-schema/configuration-sections-schema.md*.
I moved the *See also* link into the doc to possibly clear up the
mini-TOC single entry problem. Let me know if you want to go a different
way with it.

GuardRex commented:
@rpetrusha Changes made: It was nice to get the consistency addressed. :+1:

When the build is finished, I'll see if there are any :boom: on this set of updates.

Idk what to do about the headings for *docs/framework/configure-apps/file-schema/configuration-sections-schema.md*. I moved the *See also* link into the doc to possibly clear up the mini-TOC single entry problem. Let me know if you want to go a different way with it.

mairaw commented on docs/framework/configure-apps/file-schema/clear-element-for-configsections.md:
```diff
@@ -0,0 +1,101 @@
+---
```
this topic doesn't seem to sync to the TOC

mairaw commented on docs/framework/migration-guide/net-framework-4-migration-issues.md:
```diff
@@ -0,0 +1,336 @@
+  - "application compatibility"
+ms.assetid: df478548-8c05-4de2-8ba7-adcdbe1c2a60
+author: "guardrex"
+ms.author: "mariaw"
```
it should be ronpet here

GuardRex replied:
Yes ... I'll fix it.

Commit:
React to feedback

GuardRex commented:
@mairaw I added the missing topics to the TOC. Some of the TOC entries at *docs/framework/configure-apps/file-schema/toc.md* are a bit long. I also tried to organize the order ... elements at the top, alphabetical schema links at the bottom. That might need more work.

rpetrusha commented:
@GuardRex, this looks good. Sorry that I neglected it for so long. I'm going to merge now, so we'll finally incorporate it into docs.
